backend/src/main/java/com/google/capstone/UserUploadHandler.java

```diff
@@ -49,19 +48,36 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response) t
   }
   
   /**
-   * doesn't do anything other than handle the redirect from the backend back to the frontend (as of now)
+   * Gets image information from form and stores in Datastore 
+   * Note to self: store blobkey for access?
    */
   @Override
   protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
-      BlobstoreService blobstore = BlobstoreServiceFactory.getBlobstoreService();
+      String artistName = getParameter(request, "artistName", "");
+      String artTitle = getParameter(request, "artTitle", "");
+      String description = getParameter(request, "description", "");
+      long timestamp = System.currentTimeMillis();
 
-      Map<String, List<FileInfo>> uploads = blobstore.getFileInfos(request);
-      List<FileInfo> fileInfos = uploads.get("files");
+      Entity mssgEntity = new Entity("ImageInformation");
+      mssgEntity.setProperty("artistName", artistName);
+      mssgEntity.setProperty("artTitle", artTitle);
+      mssgEntity.setProperty("description", description);
+      mssgEntity.setProperty("timestamp", timestamp);
 
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+      datastore.put(mssgEntity); 
+      
       //deployment link
       //response.sendRedirect("https://igunda-isangimino-nstroupe.uc.r.appspot.com/user-uploads-gallery");
       //development link
       response.sendRedirect("https://3001-ba659410-163c-49e0-b45f-c22c5b2dc8b5.us-central1.cloudshell.dev/user-uploads-gallery");
   }
-
+  
+  private String getParameter(HttpServletRequest request, String name, String defaultValue) {
```

Comment:
Might be an option to instead send an error response and handle that in the frontend because I don't think a default value makes sense for these since they are required fields.